internal/app/machined/pkg/controllers/k8s/kubelet_spec.go

```diff
@@ -166,7 +172,7 @@ func (ctrl *KubeletSpecController) Run(ctx context.Context, r controller.Runtime
 			return fmt.Errorf("error merging arguments: %w", err)
 		}
 
-		kubeletConfig, err := NewKubeletConfiguration(cfgSpec)
+		kubeletConfig, err := NewKubeletConfiguration(cfgSpec, getStaticPodURL(ctx, r, logger))
```

Comment:
I think we shouldn't create with empty static pod URL, as this means restarting the kubelet. we can pull the HTTP server status before this call and skip reconcile if it's not ready yet